packages/leancode_lint/lib/lints/use_dedicated_media_query_methods.dart

```diff
@@ -0,0 +1,157 @@
+          errorSeverity: error.ErrorSeverity.WARNING,
+          problemMessage:
+              'Avoid using {0} to access only one property of MediaQueryData. '
+              'Using aspects of the `MediaQuery` avoids unnecessary rebuilds.\n',
+          correctionMessage: 'Use the dedicated `{1}` method instead.',
+        ),
+      );
+
+  static const _supportedGetters = {
+    'size',
+    'orientation',
+    'devicePixelRatio',
+    'textScaleFactor',
+    'textScaler',
+    'platformBrightness',
+    'padding',
+    'viewInsets',
+    'systemGestureInsets',
+    'viewPadding',
+    'alwaysUse24HourFormat',
+    'accessibleNavigation',
+    'invertColors',
+    'highContrast',
+    'onOffSwitchLabels',
+    'disableAnimations',
+    'boldText',
+    'navigationMode',
+    'gestureSettings',
+    'displayFeatures',
+    'supportsShowingSystemContextMenu',
+  };
+
+  @override
+  List<Fix> getFixes() => [_ReplaceMediaQueryOfWithDedicatedMethodFix()];
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addMethodInvocation((node) {
+      if (_isValidMediaQueryUsage(node)) {
+        return;
+      }
+
+      final replacementSuggestion = _getReplacementSuggestion(node);
+      final parent = node.parent;
+
+      if (replacementSuggestion == null || parent == null) {
+        return;
+      }
+
+      reporter.atNode(
+        parent,
+        code,
+        arguments: [node.toSource(), replacementSuggestion],
+        data: replacementSuggestion,
+      );
+    });
+  }
+
+  String? _getReplacementSuggestion(MethodInvocation node) {
+    final methodReplacement = _getReplacementMethodName(node);
+    if (methodReplacement == null) {
+      return null;
+    }
+
+    final contextVariableName = _getContextVariableName(node);
+    if (contextVariableName == null) {
+      return null;
+    }
+
+    final usedMaybe = methodReplacement.startsWith('maybe');
+    final usedGetter = _getUsedGetter(node);
+    final shouldAddQuestionMark =
+        usedMaybe && usedGetter != null && _isGrandParentPropertyAccess(node);
+
+    return 'MediaQuery.$methodReplacement($contextVariableName)${shouldAddQuestionMark ? '?' : ''}';
+  }
+
+  bool _isGrandParentPropertyAccess(MethodInvocation node) =>
+      node.parent?.parent is PropertyAccess;
+
+  String? _getContextVariableName(MethodInvocation node) =>
+      node.argumentList.arguments.firstOrNull?.toString();
+
+  String? _getReplacementMethodName(MethodInvocation node) {
+    final usedGetter = _getUsedGetter(node);
+
+    if (usedGetter == null || !_supportedGetters.contains(usedGetter)) {
+      return null;
+    }
+
+    return switch (node.methodName.name) {
+      'of' => '${usedGetter}Of',
+      'maybeOf' when usedGetter.length > 1 =>
+        'maybe${usedGetter[0].toUpperCase()}${usedGetter.substring(1)}Of',
+      'maybeOf' => 'maybe${usedGetter.toUpperCase()}Of',
```

Comment:
I don't think we need to special-case one-letter properties; always converting them will still work